### PR TITLE
chore: update `near-sdk`, `near-abi`, `borsh` version

### PR DESCRIPTION
**Pull request by dj8yfo**

### CODE REVIEW

File: integration-tests/Cargo.toml [hunk 1]
```diff
@@ -6,7 +6,7 @@ publish = false
 
 [dependencies]
 const_format = "0.2"
-near-abi = "0.3.0"
+near-abi = "0.4.0"
 
 [dev-dependencies]
 borsh = "0.9"
```
Comment (frol) [inline integration-tests/Cargo.toml:12]: Update borsh here as well

File: integration-tests/src/lib.rs [hunk 2]
```diff
@@ -109,15 +114,12 @@ macro_rules! generate_abi_fn_with {
         $crate::generate_abi_with! {
             $(Cargo: $cargo_path;)? $(Vars: $cargo_vars;)? $(Opts: $cli_opts;)?
             Code:
-            // fixme! remove after borsh fixes trait qualification
-            // fixme! https://github.com/near/borsh-rs/issues/112
```
Comment (dj8yfo) [inline integration-tests/src/lib.rs:113]: this line has set off [pr](https://github.com/near/borsh-rs/pull/244)

File: Cargo.toml [hunk 3]
```diff
@@ -37,3 +37,6 @@ pr-run-mode = "upload"
 # This should not be necessary once the following issue is addressed:
 # https://github.com/axodotdev/cargo-dist/issues/423
 allow-dirty = ["ci"]
+
+[patch.crates-io]
+borsh = { git = 'https://github.com/near/borsh-rs.git', branch = "slice_schema_container" }
```
Comment (dj8yfo) [inline Cargo.toml:42]: TODO: remove after [borsh#245](https://github.com/near/borsh-rs/pull/245) is merged

File: integration-tests/templates/_Cargo.toml [hunk 4]
```diff
@@ -21,3 +23,6 @@ opt-level = "z"
 lto = true
 debug = false
 panic = "abort"
+
+[patch.crates-io]
+borsh = { git = 'https://github.com/near/borsh-rs.git', branch = "slice_schema_container" }
```
Comment (dj8yfo) [inline integration-tests/templates/_Cargo.toml:28]: TODO: remove after [borsh#245](https://github.com/near/borsh-rs/pull/245) is merged

File: integration-tests/templates/negative/_Cargo_no_abi_feature.toml [hunk 5]
```diff
@@ -7,7 +7,9 @@ edition = "2021"
 crate-type = ["cdylib"]
 
 [dependencies]
-near-sdk = { default-features = false, version = "::sdk-version::" }
+# NOTE: uncomment line 11 and comment 12, if it's desirable to switch to published `near-sdk` version
+# near-sdk = { default-features = false, version = "::sdk-version::" }
```
Comment (frol) [inline integration-tests/templates/negative/_Cargo_no_abi_feature.toml:11]: Should it be this way?

```suggestion
# near-sdk = { default-features = false, ::sdk-version:: }
```

Comment (dj8yfo) [reply integration-tests/templates/negative/_Cargo_no_abi_feature.toml:11]: nope, `sdk-version` is `4.1.1`,
and `sdk-version-toml` is `version = "4.1.1"`

Comment (frol) [reply integration-tests/templates/negative/_Cargo_no_abi_feature.toml:11]: @dj8yfo Do we need two ways of achieving the same thing? In other files I see we use `sdk-version-toml`:

https://github.com/near/cargo-near/pull/109/files#diff-ee562e36f9098ae09c28cb061037a3122c4492b7333f6ffb188b40efa01ba9d9R11

**Review by frol — DISMISSED** [review]
This looks good, but it seems that since the ABI format got changed, we cannot cut a release of cargo-near until we cut near-sdk-rs release.

File: integration-tests/templates/negative/_Cargo_no_abi_feature.toml (hunk 5 above)
Comment (dj8yfo) [reply integration-tests/templates/negative/_Cargo_no_abi_feature.toml:11]: @frol i'll replace with `sdk-version-toml` everywhere to reduce the number of variants to express the same thing

**Review by frol — APPROVED** [review]
near-sdk-rs 5.0.0-alpha.1 is out, so we can finally merge this one

**Pull request merged**